plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/intermediate/ConstraintsPostprocessor.java

```diff
@@ -0,0 +1,63 @@
+/*******************************************************************************
+ * Copyright (c) 2019, 2020 Kiel University and others.
+ * 
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.elk.alg.layered.intermediate;
+
+import org.eclipse.elk.alg.layered.JsonDebugUtil;
+import org.eclipse.elk.alg.layered.graph.LGraph;
+import org.eclipse.elk.alg.layered.graph.LNode;
+import org.eclipse.elk.alg.layered.graph.LNode.NodeType;
+import org.eclipse.elk.alg.layered.graph.Layer;
+import org.eclipse.elk.alg.layered.options.LayeredOptions;
+import org.eclipse.elk.core.alg.ILayoutProcessor;
+import org.eclipse.elk.core.util.IElkProgressMonitor;
+
+/**
+ * Adds to each LNode the layerID and positionID that has been computed by ELKLayered.
+ * 
+ * @author cos, jet
+ */
+public final class ConstraintsPostprocessor implements ILayoutProcessor<LGraph> {
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see org.eclipse.elk.core.alg.ILayoutProcessor#process(java.lang.Object,
+     * org.eclipse.elk.core.util.IElkProgressMonitor)
+     */
+    @Override
+    public void process(LGraph graph, IElkProgressMonitor progressMonitor) {
+
+        progressMonitor.begin("Constraints Postprocessor", 1);
+        progressMonitor.log("Constraints Postprocessor started");
+        progressMonitor.log(JsonDebugUtil.createDebugGraph(graph));
```

Comment:
There's a special log method for logging graphs, I believe. Also, the JSON representation should only be generated if logging is enabled in the first place.